pyomo/contrib/incidence_analysis/interface.py

```diff
@@ -43,13 +47,12 @@
     ColPartition,
 )
 
-asl_available = False
-if scipy_available:
-    import scipy as sp
-    from pyomo.contrib.pynumero.asl import AmplInterface
-    if AmplInterface.available():
-        asl_available = True
-        from pyomo.contrib.pynumero.interfaces.pyomo_nlp import PyomoNLP
+from pyomo.contrib.pynumero.asl import AmplInterface
+asl_available = AmplInterface.available()
+if asl_available and scipy_available:
+    # Need this check due to unguarded scipy import in pyomo_nlp.py
+    # Not sure if asl_available is necessary for this import...
+    from pyomo.contrib.pynumero.interfaces.pyomo_nlp import PyomoNLP
```

Comment:
AmplInterface import seems fine on bare env. I always forget which parts of PyNumero have unguarded numpy/scipy imports...

Reply:
pynumero is a bit of a mess because so much of it has unguarded imports of numpy/scipy.  Given that we are rerunning all the tests again from scratch, I will put my edits as a PR to this PR and get everything included at once.